(GH-1267) Support project level Puppet content (#1679)
puppetlabs/bolt: documentation/directory_structure.md

Puppet tasks, plans, functions, classes and types must exist inside a Puppet module in order for Bolt to load them. Bolt loads modules by searching for module directories on the modulepath.

> By default, the modulepath includes the `modules/` and `site-modules` directories in the [Bolt project directory](bolt_project_directories.md#). The Bolt project directory itself is also loaded as a module, namespaced to the name of the directory.

Reviewer: ```suggestion
By default, the modulepath includes the `modules/` and `site-modules/` directories in the [Bolt project directory](bolt_project_directories.md#). The Bolt project directory itself is also loaded as a module, namespaced to the name specified in the `bolt.yaml`.
```

> By default, the modulepath includes the `modules/` and `site-modules` directories in the [Bolt project directory](bolt_project_directories.md#). The Bolt project directory itself is also loaded as a module, namespaced to the `project-name` specified in `bolt.yaml`.

Reviewer 2: ```suggestion
By default, the modulepath includes the `modules` and `site-modules` directories in the [Bolt project directory](bolt_project_directories.md#). The Bolt project directory itself is also loaded as a module, namespaced to the `project-name` specified in `bolt.yaml`.
```

### Where to put module code

> Modules can be developed directly in the Bolt project directory or be installed from the Puppet

Reviewer 2: ```suggestion
You can develop modules directly in the Bolt project directory or install them from the Puppet
```

> Forge into `modules/`.

Reviewer 2: ```suggestion
Forge into the `modules` directory.
```

## Modules for projects

> Modules developed to support a particular project can be developed directly in the Bolt project directory. You can use `pdk new module` to create a skeleton structure, and `bolt project init` inside a directory to make it a Bolt project directory.

Reviewer 2: ```suggestion
Modules developed to support a particular project can be developed directly in the Bolt project directory. You can use `pdk new module` to create a skeleton structure, and `bolt project init` inside a directory to make it a Bolt project directory.

**Note**:  To use the `pdk` command, you must [install the Puppet Development Kit](https://puppet.com/docs/pdk/1.x/pdk_install.html) 
```

Follow these tips for managing standalone modules:

-   Add `modules/*` to `.gitignore` of your project to prevent accidentally committing standalone modules.

> -   When you run tasks and plans within a project directory the modulepath is searched in order for modules containing Bolt content. The Bolt project directory itself is loaded as a module, and the default modulepath is `<project directory>/modules:<project directory>/site-modules`. So if you have a module in both the `modules` and `site-modules` directories, the version in `modules` will actually be used.

Reviewer: ```suggestion
-   When you run tasks and plans within a project directory the modulepath is searched in order for modules containing Bolt content. The Bolt project directory itself is loaded as a module, and the default modulepath is `<project directory>/modules:<project directory>/site-modules`. So if you have a module in both the `modules` and `site-modules` directories, the version in `modules` will be used.
```
